internal/cli/base.go

```diff
@@ -354,47 +300,47 @@ func (c *baseCommand) Init(opts ...Option) error {
 	c.variables = vars
 
 	// Create our client
-	if baseCfg.Client {
+	if !baseCfg.NoClient {
 		c.project, err = c.initClient(nil)
 		if err != nil {
 			c.logError(c.Log, "failed to create client", err)
 			return err
 		}
 	}
 
-	// Validate remote vs. local operations.
-	if !baseCfg.AppOptional {
-		if c.flagRemote && c.refApp == nil {
-			if c.cfg == nil || c.cfg.Runner == nil || !c.cfg.Runner.Enabled {
-				err := errors.New(
-					"The `-remote` flag was specified but remote operations are not supported\n" +
-						"for this project.\n\n" +
-						"Remote operations must be manually enabled by using setting the 'runner.enabled'\n" +
-						"setting in your Waypoint configuration file. Please see the documentation\n" +
-						"on this setting for more information.")
-				c.logError(c.Log, "", err)
-				return err
-			}
+	// Start a local runner if necessary
+	if baseCfg.RunnerRequired && !c.remoteOpPreferred() {
+		_, err := c.startLocalRunner()
+		if err != nil {
+			c.logError(c.Log, "failed to start a local runner", err)
+			return err
 		}
 	}
 
-	// If this is a single app mode then make sure that we only have
-	// one app or that we have an app target.
-	if baseCfg.AppTargetRequired {
-		if c.refApp == nil {
-			if len(c.cfg.Apps()) != 1 {
-				c.ui.Output(errAppModeSingle, terminal.WithErrorStyle())
-				return ErrSentinel
-			}
+	return nil
+}
 
-			c.refApp = &pb.Ref_Application{
-				Project:     c.cfg.Project,
-				Application: c.cfg.Apps()[0],
-			}
-		}
+// startLocalRunner starts a local runner, and adds its id to the grpc context.
+func (c *baseCommand) startLocalRunner() (*runner.Runner, error) {
```

Comment:
We need to break this out of the project client's init sequence, because in the near future we will need to _use_ the project client to determine if we should start the local runner.